src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+import os
+import uuid
+import logging
+import numpy as np
+from time import sleep
+from pathlib import Path
+from qt3utils.errors import  QT3Error
+from typing import Any, Tuple, List, Union
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+try:
+    import clr
+
+    lf_root = Path(os.environ['LIGHTFIELD_ROOT'])
+    automation_path = lf_root / 'PrincetonInstruments.LightField.AutomationV4.dll'
+    addin_path = lf_root / 'AddInViews' / 'PrincetonInstruments.LightFieldViewV4.dll'
+    support_path = lf_root / 'PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(str(addin_path))
+    automation_class = clr.AddReference(str(automation_path))
+    support_class = clr.AddReference(str(support_path))
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String, Int32, Int64, IntPtr, Double
+    from System.IO import FileAccess
+except KeyError as e:
+    logger.error(f"KeyError {e} during import")
+except ImportError as e:
+    logger.error(f"Unable to import packages: {e}")
+
+
+class LightfieldApplicationManager:
+    def __init__(self, visible: bool) -> None:
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def experiment(self) -> Any:
+        return self._experiment
+
+    def __del__(self) -> None:
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        #NOTE: May need to call '.Dispose()' here again as qt3scan may fail to call 'close'
+        self._automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+    def set(self, setting: str, value: Any) -> None:
+        """
+        Helper function for setting experiment parameters with basic value
+        checking.
+        """
+        if self.experiment.Exists(setting):
+            if self.experiment.IsValid(setting, value):
+                self.experiment.SetValue(setting, value)
+            else:
+                # TODO: might need to add a proper exception
+                logger.error(f'Invalid value: {value} for setting: {setting}.')
+        else:
+            # TODO: might need to add a proper exception
+            logger.error(f'Invalid setting: {setting}.')
+
+    def get(self, setting: Any) -> Any:
```

Comment:
using Any as type hint for the setting argument seems a bit too vague for me. Can you add a docstring explaining what settings means, and potentially change the type hint to better reflect the potential input value types?

Reply:
I think `Any` is actually reasonable here since this is a generic wrapper around getting any parameter in LF. However, a docstring and links to Princeton documentation about what values a user can "get" would be helpeful. Also, instead of `Any`, it could be a Union[str, float, int] if specifying python types, or it could be Union[System.Int32, System.Double, System.String] etc.

Reply:
wouldn't the argument be a string for the getter? Not sure how LF works. If it's too many things, then Any is fine. 

I agree that using Any for the return value is probably wise.

Reply:
Not necessarily. I think it is a variety of things